bindings/python/gnucash_core.py

```diff
@@ -117,6 +117,15 @@ def __init__(self, book_uri=None, ignore_lock=False, is_new=False,
                 self.destroy()
                 raise
 
+    def __enter__(self):
+        return self
+
+    def __exit__(self, exc_type, exc_value, traceback):
+        # Roll back changes on exception by not calling save. Only works for XMl backend.
```

Comment:
To avoid unneeded saves you can first test if the book is dirty before calling save. As sql based books are never in a dirty state that also avoids calling save on sql based books.

Reply:
My understanding is that the underlying code is already checking for a dirty book:
 Session.save() is qof_session_save, which calls QofSessionImpl::save, which does nothing if !qof_book_session_not_saved.

Reply:
That's true indeed. So why the comment this would only work for XML backend ?

Reply:
My understanding is that qof_session_save is a noop for sql backends, so the rollback would not occur for sql backends. Is that correct? I'm not an expert on the codebase.

Is there a better way I can word that comment to be clear?